public/static/sdk/jsmr-0.1.js

```diff
@@ -1,104 +1,155 @@
         initializing = true;
         var prototype = new this();
         initializing = false;
-     
+
         // Copy the properties over onto the new prototype
         for (var name in prop) {
             // Check if we're overwriting an existing function
             prototype[name] = typeof prop[name] == "function" &&
-                typeof _super[name] == "function" && fnTest.test(prop[name]) ?
-                (function(name, fn){
-                    return function() {
+            typeof _super[name] == "function" && fnTest.test(prop[name]) ?
+                (function (name, fn) {
+                    return function () {
                         var tmp = this._super;
-                     
+
                         // Add a new ._super() method that is the same method
                         // but on the super-class
                         this._super = _super[name];
-                     
+
                         // The method only need to be bound temporarily, so we
                         // remove it when we're done executing
                         var ret = fn.apply(this, arguments);
                         this._super = tmp;
-                     
+
                         return ret;
                     };
                 })(name, prop[name]) :
                 prop[name];
         }
-     
+
         // The dummy class constructor
         function Class() {
             // All construction is actually done in the init method
-            if ( !initializing && this.init )
+            if (!initializing && this.init)
                 this.init.apply(this, arguments);
         }
-     
+
         // Populate our constructed prototype object
         Class.prototype = prototype;
-     
+
         // Enforce the constructor to be what we expect
         Class.prototype.constructor = Class;
- 
+
         // And make this class extendable
         Class.extend = arguments.callee;
-     
+
         return Class;
     };
 })();
 
+
+
 var JsMr = Class.extend({
     init: function () {
+
         this.registered = false;
         this.client_id = null;
         this.last_call = 0;
         this.current_task = null;
+        this.current_output = "";
+        this.current_state = null;
+        this.step_list = [];
+
         this.beat_interval_obj = null;
+        this.aws_sdk = 'https://sdk.amazonaws.com/js/aws-sdk-2.1.21.min.js';
         this.options = {
             beat_interval: 60000, // one minute
             auth_token: null,
             server_path: '',
             debug: true,
-            jquery_cdn: 'https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js',
-        }
+            jquery_cdn: 'https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js'
+        };
 
         // Start the SDK
         this.loadAndRegister();
     },
 
     /**
+     * Upload the output and state to AWS S3
+     *
+     */
+    upload_data: function () {
+
+        var s3 = new AWS.S3();
```

Comment:
It might be better to create a global S3 object on the class to avoid creating the object for each task.